app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -66,16 +66,23 @@ def update_os_from_facts
 
         def self.find_with_expiring_pools(_key, _operator, days_from_now)
           minimum_pool_date = (Date.today + days_from_now.to_i.days).strftime('%Y-%m-%d')
+          host_ids = self.joins(:pools).where(["katello_pools.end_date < ?", minimum_pool_date]).distinct.ids
+          if host_ids.blank?
+            return {
+              :conditions => "katello_pools.end_date < ?",
+              :parameter => [minimum_pool_date],
+              :joins => :pools
+            }
+          end
```

Comment:
`scoped_search` was causing a SQL syntax error whenever the host_ids returned zero hosts.  This workaround looks weird but seems to work fine.

You may be asking, why not just return this no matter what?  It's because `scoped_search` doesn't allow any way to do `SELECT DISTINCT`, so it was giving me duplicate Host objects and thus, duplicate report rows.  I work around that by doing `.distinct` myself and then just giving scoped_search the ID's.